File: solutions/javascript/scrabble-score/1/scrabble-score.js
//
// This is only a SKELETON file for the 'Scrabble Score' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const score = (word) => {
  var score = 0;
  for (var ch of word) {
    if ('A, E, I, O, U, L, N, R, S, T'.indexOf(ch.toUpperCase())>=0) score += 1;
    if ('D, G'.indexOf(ch.toUpperCase())>=0) score += 2;
    if ('B, C, M, P'.indexOf(ch.toUpperCase())>=0) score += 3;
    if ('F, H, V, W, Y'.indexOf(ch.toUpperCase())>=0) score += 4;
    if ('K'.indexOf(ch.toUpperCase())>=0) score += 5;
    if ('J, X'.indexOf(ch.toUpperCase())>=0) score += 8;
    if ('Q, Z'.indexOf(ch.toUpperCase())>=0) score += 10;
  }
  return score;
  //throw new Error('Remove this statement and implement this function');
};

File: solutions/go/scrabble-score/1/scrabble_score.go
package scrabble
import "strings"

func Score(word string) (int) {
	
  var lcw = strings.ToLower(word) 
	var s int = 0;
	for i := range lcw {
	  switch lcw[i] {
    case 'a','e','i','o','u','l','n','r','s','t':
			s++;
    case 'd','g':
			s += 2;
    case 'b','c','m','p':
			s += 3;
    case 'f','h','v','w','y':
			s += 4;
    case 'k':
			s += 5;
    case 'j','x':
			s += 8;
    case 'q','z':
			s += 10;
		}
	}
	return s

}
